### DISCUSSION TITLE
What's the correct way to use chokidar to track an everchanging set of files and directories?

### DISCUSSION THREAD

**Discussion by Beedeebee** [paulmillr/chokidar#1256] [Q&A]
I'm developing a programming language and need to write a build-tool able to recompile what changes, whenever it changes: a tool that computes dependencies recursively starting from an entry file, tracks whenever a file changes and rebuild what's needed.

The dependencies can use wildcards, so I need to watch both files and directories (either recursively or not), for instance:
- `file.in`: watch that one file.
- `plugins/*.in`: watch the `plugins/` directory and every `.in` file in it.
- `imgs/**/*.{jpg,png}`: watch the `imgs/` directory recursively and every JPG/PNG file in there.

All the logic is already written, but my current file-watcher has too many issues and I need to rewrite it.

I'm wondering:

1. Is chokidar the right tool for it, or would you recommend looking elsewhere?
2. What's the best way to use chokidar to watch for files and directories in my case? Should I...
   - Run a single watcher on the program's directory, listen to `"all"` events and filter them by myself?
   - `watcher.add()` and `watcher.unwatch()` the files and directories I need?
   - Create and destroy new watchers for the files and directories I need?

Is there any project that uses chokidar for this purpose that I can look at?

**Comment by paulmillr** [comment]
> Is chokidar the right tool for it

Yes.
